src/main/java/org/example/CalendarManagement/thriftclients/implementation/MeetingServiceClientImpl.java

```diff
@@ -0,0 +1,182 @@
+        try( TTransport transport = new THttpClient("http://localhost:" + 8080 + "/checkEmployeeAvailability")) {
+            transport.open();
+
+            TProtocol protocol = new TBinaryProtocol(transport);
+
+            MeetingSvc.Client client = new MeetingSvc.Client(protocol);
+
+            List<String> listOfEmployeeNotAvailable = client.checkEmployeeAvailability(employeeAvailabilityDataRequest);
+
+            transport.close();
+
+            return listOfEmployeeNotAvailable;
+        }catch (TException exception)
+        {
+            throw  new RuntimeException(exception.getMessage());
+        }
+    }
+
+    @Override
+    public String addMeetingDetails(MeetingDetails meetingDetails) {
+        try( TTransport transport = new THttpClient("http://localhost:" + 8080 + "/addMeetingDetails")) {
+            transport.open();
+
+            TProtocol protocol = new TBinaryProtocol(transport);
+
+            MeetingSvc.Client client = new MeetingSvc.Client(protocol);
+
+            String thriftResponse = client.addMeetingDetails(meetingDetails);
+
+            transport.close();
+
+            return thriftResponse;
+        }catch (TException exception)
+        {
+            throw  new RuntimeException(exception.getMessage());
+        }
+    }
+
+    @Override
+    public boolean addEmployeeMeetingStatus(List<EmployeeStatusDataRequest> employeeStatusDataRequest) {
+        try( TTransport transport = new THttpClient("http://localhost:" + 8080 + "/addEmployeeMeetingStatus")) {
+            transport.open();
+
+            TProtocol protocol = new TBinaryProtocol(transport);
+
+            MeetingSvc.Client client = new MeetingSvc.Client(protocol);
+
+            Boolean thriftResponse = client.addEmployeeMeetingStatus(employeeStatusDataRequest);
+
+            transport.close();
+
+            return thriftResponse;
+        }catch (TException exception)
+        {
+            throw  new RuntimeException(exception.getMessage());
+        }
+    }
+
+    @Override
+    public Integer findFreeMeetingRoom(FindFreeMeetingRoomDataRequest findFreeMeetingRoomDataRequest) {
+        try( TTransport transport = new THttpClient("http://localhost:" + 8080 + "/findFreeMeetingRoom")) {
+            transport.open();
+
+            TProtocol protocol = new TBinaryProtocol(transport);
+
+            MeetingSvc.Client client = new MeetingSvc.Client(protocol);
+
+            Integer thriftResponse = client.findFreeMeetingRoom(findFreeMeetingRoomDataRequest);
+
+            transport.close();
+
+            return thriftResponse;
+        }catch (TException exception)
+        {
+            throw  new RuntimeException(exception.getMessage());
+        }
+    }
+
+    @Override
+    public boolean meetingRoomAvailable(MeetingRoomAvailableDataRequest meetingRoomAvailableDataRequest) {
+        try( TTransport transport = new THttpClient("http://localhost:" + 8080 + "/meetingRoomAvailable")) {
```

Comment:
see if we can move the common code into a separate method?
can we reuse the same client, and create once and reuse till application shutdown, low priority?